hazelcast/src/main/java/com/hazelcast/map/impl/operation/PutAllBackupOperation.java

```diff
@@ -52,11 +54,17 @@ public PutAllBackupOperation() {
     public void run() {
         int partitionId = getPartitionId();
         recordStore = mapService.getMapServiceContext().getRecordStore(partitionId, name);
+        boolean wanEnabled = mapContainer.getWanReplicationPublisher() != null && mapContainer.getWanMergePolicy() != null;
         for (int i = 0; i < entries.size(); i++) {
             final RecordInfo recordInfo = recordInfos.get(i);
             final Map.Entry<Data, Data> entry = entries.get(i);
             final Record record = recordStore.putBackup(entry.getKey(), entry.getValue());
             Records.applyRecordInfo(record, recordInfo);
+            if (wanEnabled) {
+                final Data dataValueAsData = mapService.getMapServiceContext().toData(entry.getValue());
```

Comment:
Why are you retrieving the serviceContext in the loop? It makes the code very hard to read.